### ISSUE TITLE
RestClient should expose a wither to specify poll interval for long running operations.

### ISSUE THREAD

**Issue by anuchandy** [Azure/autorest-clientruntime-for-java#304]

**Comment by daschult** [comment]
RestClient wouldn't be the correct place to do this since long running operations occur in the Azure layer and RestClient appears in the REST layer (underneath the Azure layer). Not to mention I'm removing RestClient in favor of an HttpPipeline class. What's the scenario that made you think of this?

**Comment by anuchandy** [comment]
@daschult You are right, agree RestClient is one layer below, so it's not right place. 

The scenario is - customers using the fluent layer want control over the default polling interval of LROs. In the fluent layer, we allow composing create-ables for resources and calling create on the root resource will cause creating all the composed create-ables underneath `implicitly`. These `implicit`  creations are LROs. Customers are looking for one place to configure this.

In fluent layer we could expose something like:

```java
            Azure azure = Azure.configure()
                    .withLongRunningOperationPollingInterval(timespan)
                    .withLogLevel(LogLevel.BODY_AND_HEADERS)
                    .authenticate(credFile)
                    .withDefaultSubscription();
```

Is it possible to expose a property in the generated azure service client where we can set LRO default interval that can be passed to the runtime LRO abstraction? so that when fluent layer creates inner service clients it can configure it.

**Comment by RikkiGibson** [comment]
This is probably worthwhile for the `TEntity create()` overloads which do polling automatically. I figure that we would provide this parameter similarly to how we provide AzureEnvironment by making it a property on the base AzureServiceClient. Generated constructors for AzureServiceClient subclasses would have this additional parameter.

**Comment by RikkiGibson** [comment]
Action items:

- Add `longRunningOperationPollInterval` property and constructor parameter to AzureServiceClient
- Add argument to generated constructors
- Add argument to `AzureProxy.create()`
